core/client/fs/src/main/java/alluxio/client/block/policy/RoundRobinPolicy.java

```diff
@@ -35,10 +36,11 @@
 @NotThreadSafe
 public final class RoundRobinPolicy implements BlockLocationPolicy {
   private List<BlockWorkerInfo> mWorkerInfoList;
-  private int mIndex;
+  private static AtomicInteger mIndex = new AtomicInteger();
```

Comment:
Does it make better sense if you just make this class a singleton?

Reply:
Basically using a static field to keep the state is a weird design, singleton is a better choice.